core/includes/file.inc

```diff
@@ -1579,7 +1579,8 @@ function file_save_upload($form_field_name, $validators = array(), $destination
   else {
     // No validator was provided, so add one using the default list.
     // Build a default non-munged safe list for file_munge_filename().
-    $extensions = 'jpg jpeg gif png txt doc xls pdf ppt pps odt ods odp docx docm pptx pptm xlsx xlsm';
+    $extensions = implode(' ', image_get_supported_extensions()) . ' txt doc xls pdf ppt pps odt ods odp docx docm pptx pptm xlsx xlsm';
```

Comment:
@herbdool it would be nice to move all these non-image extensions to a function or a constant instead (`file_get_default_allowed_extensions()` or something like that), that also returns them as an array (ready to be iterated through) + also reorder it to something like this (group similar types together):

`txt pdf doc docx docm xls xlsx xlsm pps ppt pptx pptm odt ods odp`

Reply:
That sounds like a new issue.

Reply:
Agreed. Done: https://github.com/backdrop/backdrop-issues/issues/6054